packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/src/models/models.ts

```diff
@@ -0,0 +1,502 @@
+  readonly keyVaultUrl?: string;
+}
+
+/** The status of the current operation. */
+/** "Succeeded", "Failed", "Canceled", "Provisioning", "Updating", "Deleting", "Accepted" */
+export type ProvisioningState = string;
+/** The data type state */
+/** "Enabled", "Disabled" */
+export type ControlState = string;
+
+/** Encryption key details. */
+export interface EncryptionKeyDetails {
+  /** The Uri of the key vault. */
+  keyVaultUri: string;
+  /** The name of the key vault key. */
+  keyName: string;
+  /** The version of the key vault key. */
+  keyVersion: string;
+}
+
+/** Data Product Network rule set */
+export interface DataProductNetworkAcls {
+  /** Virtual Network Rule */
+  virtualNetworkRule: VirtualNetworkRule[];
+  /** IP rule with specific IP or IP range in CIDR format. */
+  ipRules: IPRules[];
+  /** The list of query ips in the format of CIDR allowed to connect to query/visualization endpoint. */
+  allowedQueryIpRangeList: string[];
+  /** Default Action */
+  defaultAction: DefaultAction;
+}
+
+/** Virtual Network Rule */
+export interface VirtualNetworkRule {
+  /** Resource ID of a subnet */
+  id: string;
+  /** The action of virtual network rule. */
+  action?: string;
+  /** Gets the state of virtual network rule. */
+  state?: string;
+}
+
+/** IP rule with specific IP or IP range in CIDR format. */
+export interface IPRules {
+  /** IP Rules Value */
+  value?: string;
+  /** The action of virtual network rule. */
+  action: string;
+}
+
+/** Specifies the default action of allow or deny when no other rules match. */
+/** "Allow", "Deny" */
+export type DefaultAction = string;
+
+/** ManagedResourceGroup related properties */
+export interface ManagedResourceGroupConfiguration {
+  /** Name of managed resource group */
+  name: string;
+  /** Managed Resource Group location */
+  location: string;
+}
+
+/** Details of Consumption Properties */
+export interface ConsumptionEndpointsProperties {
+  /** Ingestion url to upload the data. */
+  readonly ingestionUrl?: string;
+  /** Resource Id of ingestion endpoint. */
+  readonly ingestionResourceId?: string;
+  /** Url to consume file type. */
+  readonly fileAccessUrl?: string;
+  /** Resource Id of file access endpoint. */
+  readonly fileAccessResourceId?: string;
+  /** Url to consume the processed data. */
+  readonly queryUrl?: string;
+  /** Resource Id of query endpoint. */
+  readonly queryResourceId?: string;
+}
+
+/** The properties of the managed service identities assigned to this resource. */
+export interface ManagedServiceIdentity {
+  /** The Active Directory tenant id of the principal. */
+  readonly tenantId?: string;
+  /** The active directory identifier of this principal. */
+  readonly principalId?: string;
+  /** The type of managed identity assigned to this resource. */
+  type: ManagedServiceIdentityType;
+  /** The identities assigned to this resource by the user. */
+  userAssignedIdentities?: Record<string, UserAssignedIdentity>;
```

Comment:
The modular would be generated as Record.

Reply:
we should generate the same as RLC, I think it's a bug.